### PR TITLE
hdmap_common: improve search efficiency

### PR DESCRIPTION
**Pull request by lx18233184051**

### CODE REVIEW

File: modules/map/hdmap/hdmap_common.cc [hunk 1]
```diff
@@ -308,8 +308,15 @@ double LaneInfo::GetWidthFromSample(
   }
   int low = 0;
   int high = static_cast<int>(samples.size());
+  int mid;
   while (low + 1 < high) {
-    const int mid = (low + high) / 2;
+    double tmp = samples[high].first - samples[low].first;
```
Comment (changsh726) [inline modules/map/hdmap/hdmap_common.cc:313]: Seems like `samples[samples.size()]` will out of bounds

File: modules/dreamview/backend/simulation_world/BUILD [hunk 2]
```diff
@@ -38,7 +38,7 @@ cc_library(
 
 cc_test(
     name = "simulation_world_service_test",
-    size = "small",
+    size = "large",
```
Comment (storypku) [inline modules/dreamview/backend/simulation_world/BUILD:41]: Could you please note the reason for the change here?

File: modules/map/hdmap/hdmap_common.cc (hunk 1 above)
Comment (storypku) [reply modules/map/hdmap/hdmap_common.cc:313]: > Seems like `samples[samples.size()]` will out of bounds

Not "seems like",  high=N will surely make visit to the element out of bounds.

File: modules/planning/reference_line/BUILD [hunk 3]
```diff
@@ -127,7 +127,7 @@ cc_test(
 
 cc_test(
     name = "discrete_points_reference_line_smoother_test",
-    size = "small",
+    size = "large",
```
Comment (storypku) [inline modules/planning/reference_line/BUILD:130]: Ditto. Please justify why "small" or "medium" is not enough ?

File: modules/dreamview/backend/simulation_world/BUILD (hunk 2 above)
Comment (lx18233184051) [reply modules/dreamview/backend/simulation_world/BUILD:41]: modify hdmap_common.cc, those tests become timeout, maybe those used  cache before.

File: modules/planning/reference_line/BUILD (hunk 3 above)
Comment (lx18233184051) [reply modules/planning/reference_line/BUILD:130]: I have tested, at least two minutes

File: modules/map/hdmap/hdmap_common.cc [hunk 4]
```diff
@@ -308,8 +308,16 @@ double LaneInfo::GetWidthFromSample(
   }
   int low = 0;
   int high = static_cast<int>(samples.size());
+  int mid;
   while (low + 1 < high) {
-    const int mid = (low + high) / 2;
+    double tmp = samples[high - 1].first - samples[low].first;
+    if (tmp > 0) {
+      mid = low + (s - samples[low].first) /
+                      (samples[high - 1].first - samples[low].first) *
```
Comment (changsh726) [inline modules/map/hdmap/hdmap_common.cc:316]: why not use `tmp` instead

**Pull request closed without merging**